app/models/user.rb

```diff
@@ -274,6 +274,24 @@ def reset_api_key
     self.update(api_key: Base64.encode64(md5.to_s).strip)
   end
 
+
+  def visible_assessments(type: nil, assessment_id: nil)
```

Comment:
Because you're using the `@current_user.visible_assessments(assessment_id: some_id).exists?` pattern a lot. It might simplify things if you add a policy function in the assignments and grade_entry_form policy:

```ruby
def visible?
  user.visible_assessments(assessment_id: record.id).exists?
end
```

and then you can call as a policy with `allowed_to? :view?`